warehouse/query-core/src/main/java/datawave/query/jexl/lookups/FieldNameLookup.java

```diff
@@ -84,33 +84,35 @@ public IndexLookupMap lookup(ShardQueryConfiguration config, ScannerFactory scan
         ScannerSession bs = null;
         
         try {
-            
-            for (String term : terms) {
-                
-                Set<Range> ranges = Sets.newHashSet();
-                
-                ranges.add(ShardIndexQueryTableStaticMethods.getLiteralRange(term));
-                if (limitToTerms) {
-                    log.trace("Creating configureTermMatchOnly");
-                    bs = ShardIndexQueryTableStaticMethods.configureTermMatchOnly(config, scannerFactory, config.getIndexTableName(), ranges,
-                                    Collections.singleton(term), Collections.emptySet(), false, true);
-                } else {
-                    log.trace("Creating configureLimitedDiscovery");
-                    bs = ShardIndexQueryTableStaticMethods.configureLimitedDiscovery(config, scannerFactory, config.getIndexTableName(), ranges,
-                                    Collections.singleton(term), Collections.emptySet(), false, true);
-                }
-                /**
-                 * Fetch the limited field names for the given rows
-                 */
-                for (Text field : fields) {
-                    bs.getOptions().fetchColumnFamily(field);
+
+            if (!fields.isEmpty()) {
+                for (String term : terms) {
+
+                    Set<Range> ranges = Sets.newHashSet();
```

Comment:
new HashSet(terms.size()) to create a set with expected size.

Reply:
+1